config/config_mgmt.py

```diff
@@ -385,6 +389,23 @@ def validateConfigData(self):
         self.sysLog(msg='Data Validation successful')
         return True
 
+    """
+    Based on the list of Ports, create a dict to shutdown port, update Config DB.
+    Input: [] of ports.
+    """
+    def shutdownIntf(self, ports):
+        """ shut down all the interfaces before deletion """
+        shutDownConf = dict(); shutDownConf["PORT"] = dict()
+        for intf in ports:
+            if not interface_name_is_valid(intf):
```

Comment:
Why do we need to check this again here? When we receive delete and add ports from CLI, aren't they already validated?

Reply:
yeah, except Parent port, nothing else is validated before.

Reply:
ok, then all of it should be validated in one place. I feel when something comes to this function, it should all be validated. Just like the way we validate CLI command options, breakout mode, root port name in pyclick, list of add ports and delete ports should also be verified at the same time! could you please explain me the reason for verifying add and delete port lists in the first place? Shouldn't we assume that they are always correct as they are being generated from platform.json? @samaity : Could you please comment on my assumption?

Reply:
This seems a good idea to validated all del_ports in PY Click itself instead of config_mgmt.py, because we should call the breakOut API only after this validation, I will move the code to main.py.
Validation is for configDb, we have port info from platform.json but we need to validate it in configDB as well that PORT really exists.

Reply:
Done